_posts/2019-05-10-server-rendering-responsively.md

```diff
@@ -0,0 +1,189 @@
+---
+layout: epic
+title: "Server-Rendering Responsively"
+date: "2019-04-22"
+author: [steve-hicks]
+categories: [react, html, web]
+---
+
+We use server-side rendering (SSR) to deliver every page you hit on [artsy.net](https://artsy.net). We decided on
+using SSR for many reasons, amongst them performance. (TODO: Maybe more about why we chose SSR, but I don't know
+much about that history.)
+
+We've also built our site using responsive design, so you get a browsing experience optimized for your device.
+
+Combining SSR and responsive design is a non-trivial problem. There are many concerns to manage, and they are
+sometimes in conflict with each other. We server render for performance reasons, but we also want to be sure our
+app is performant when your browser takes over, all while optimizing accessibility and SEO.
+
+This article describes the tools we use on [artsy.net](https://artsy.net) to combine SSR and responsive design.
+
+<!-- more -->
+
+## Tool 1: [`styled-system`](https://styled-system.com)
+
+We handle the majority of responsive styling differences with
+[`styled-system`](https://styled-system.com/responsive-styles). This has been a really great addition to our
+toolbox. Here's a component that would render a `div` (`Box`) with a width of 50% at our mobile breakpoint, 75% at
+our tablet breakpoint, and 100% for anything larger:
+
+```xml
+<Box width={["50%", "75%", "100%"]}>
+  ...
+</Box>
+```
+
+As developers, we love this experience. We can make subtle differences to components across breakpoints with very
+little code and effort.
+
+We're working on another blog post that digs deeper into how we use `styled-system` within
+[our design system](https://palette.artsy.net/). You can also
+[poke around our source](https://github.com/artsy/reaction/blob/f3dabb884d616c5c42bbd47b1432dc2a9456b8ca/src/Apps/Search/Components/SearchResultsSkeleton/Header.tsx#L6)
+to see how much we've embraced `styled-system`'s responsive styles. (In that example, `pr` means `padding-right`
+and `pl` means `padding-left`.)
```

Comment:
It might be good to elaborate a bit more on styled-system's responsive properties. `width` is a good example, but _everything_ is responsive; adding another example to fill out this section could work well. 

Also, maybe hold on mentioning the additional post, and instead talk a bit about Palette and our use of styled-system?  It seems like with the mention of a future post we're jumping away a tad too quickly from this post. Could pull some material from the docs: https://styled-system.com/responsive-styles. 

Might also be worth mentioning that "responsive props" is an idea that's starting to catch on (see [facepaint](https://github.com/emotion-js/facepaint)) and that it will work with any css-in-js library.

Reply:
Good idea - I will work on this.

Reply:
I added a "real life example", and got more detailed about responsive props. https://github.com/artsy/artsy.github.io/pull/566/files#diff-a879e573d311218fd7eec7da14d776f5R36